packages/@aws-cdk/core/lib/asset-staging.ts

```diff
@@ -15,6 +15,50 @@ import { Stage } from './stage';
 // eslint-disable-next-line
 import { Construct as CoreConstruct } from './construct-compat';
 
+/**
+ * A simple cache class.
+ *
+ * Must be declared at the top of the file because we're going to use it statically in the
+ * AssetStaging class.
+ */
+class Cache<A> {
+  private cache = new Map<string, A>();
+
+  /**
+   * Clears the cache
+   */
+  public clear() {
+    this.cache.clear();
+  }
+
+  /**
+   * Get a value from the cache or calculate it
+   */
+  public obtain(cacheKey: string, calcFn: () => A): A {
+    const old = this.cache.get(cacheKey);
+    if (old) { return old; }
+
+    const noo = calcFn();
```

Comment:
Missing several `oooooooooo`s. http://nooooooooooooooo.com/

Maybe to remove the pronunciation ambiguity:
```suggestion
    const calcValue = calcFn();
```

And `old` can become `cachedValue`?

Reply:
You are right, I was trying to be cute